File: Dictionaries and Sets/1189. Maximum Number of Balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        d = {"b": 0, "a": 0, "l": 0, "o": 0, "n": 0}

        for c in text:
            if c in d:
                d[c] += 1
        
        return min(d["b"], d["a"], d["l"] // 2, d["o"] // 2, d["n"])

# Time: O(n)
# Space: O(1)

File: Dictionaries and Sets/1189. Maximum Number of Balloons/1189. Maximum Number of Balloons.js
/**
 * @param {string} text
 * @return {number}
 */
var maxNumberOfBalloons = function (text) {
  const balloon = { b: 0, a: 0, l: 0, o: 0, n: 0 };
  for (let c of text) {
    if (c in balloon) balloon[c]++;
  }
  return Math.min(
    balloon["b"],
    balloon["a"],
    Math.floor(balloon["l"] / 2),
    Math.floor(balloon["o"] / 2),
    balloon["n"]
  );
};

// Time: O(n)
// Space: O(1)
